app/main.py

```
@@ -291,6 +291,20 @@ async def lifespan(_app: FastAPI):
+    # owns a multiprocessing SimpleQueue/Event/Lock; if their worker
+    # threads are still alive at interpreter exit, resource_tracker
+    # reports the SemLocks as "leaked semaphore objects". complete()
+    # drains pending records, remove() joins the workers and frees the
+    # primitives deterministically. Nothing may log through loguru after
+    # this point, which is why it sits at the very end of shutdown.
+    try:
+        logger.info("Shutdown complete; flushing log queues.")
+        await logger.complete()
+        await asyncio.to_thread(logger.remove)
+    except Exception:  # noqa: BLE001 - never block process exit on logging
+        pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)